components/backup/src/endpoint.rs

```diff
@@ -575,13 +573,20 @@ impl<E: Engine, R: RegionInfoProvider> Endpoint<E, R> {
         &self,
         prs: Arc<Mutex<Progress<R>>>,
         request: Request,
-        tx: mpsc::Sender<(BackupRange, Result<BackupRes>)>,
+        tx: UnboundedSender<BackupResponse>,
     ) {
         let start_ts = request.start_ts;
+        let end_ts = request.end_ts;
         let backup_ts = request.end_ts;
         let engine = self.engine.clone();
         let db = self.db.clone();
         let store_id = self.store_id;
+        let limiter = request.limiter.clone();
+        let pool_size = self.pool.borrow().size as f64;
+        let speed_limit = limiter.speed_limit();
+        if speed_limit.is_finite() {
+            limiter.set_speed_limit(speed_limit / pool_size);
```

Comment:
It does not resolve the rate limiter issue, as long as BR sends multiple backup requests concurrently TiKV could backup at the speed = (ratelimit * all requests).